File: csharpProblems/quadraticEquation/quadraticEquation.cs
using System;

namespace QuadraticEquation;


/// <summary>
/// Calculates the root of the quadratic equation given a, b, and c
/// </summary>
/// <param name="a">A in the quadratic equation</param>
/// <param name="b">B in the quadratic equation</param>
/// <param name="c">C in the quadratic equation</param>
/// <returns>Returns the resultant of the quadratic equation given a, b, and c</returns>
public static int quadraticEquation(int a, int b, int c)
{

    const int upperHalf = (-1 * b) + System.Math.Sqrt((b * b) - 4 * a * c);
    const int lowerHalf = 2 * a;
    return upperHalf / lowerHalf;

}

File: ts/quadraticEquation/quadraticEquation.ts
/**
 * Calculates the root given a, b, and c in the quadratic equation
 *
 * @param a - A in the quadratic equation
 * @param b - B in the quadratic equation
 * @param c - C in the quadratic equation
 * @returns The root value given a, b, and c in the quadratic equation
 */
const quadraticEquationFunc = (a: number, b: number, c: number): number => {
  const upperHalf: number = -b + Math.sqrt(b * b - 4 * a * c);
  const lowerHalf: number = 2 * a;
  return upperHalf / lowerHalf;
};
